ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerTestFramework.java

```diff
@@ -0,0 +1,476 @@
+          startDeleteBlob(nextOp.nonBlockingRouter, opInfo);
+          break;
+        case AWAIT_CREATION:
+          startAwaitCreation(opInfo);
+          break;
+        case AWAIT_DELETION:
+          startAwaitDeletion(opInfo);
+          break;
+      }
+    }
+  }
+
+  /**
+   * Used to specify operations to perform on a blob in an operation chain
+   */
+  enum OperationType {
+    /**
+     * PutBlob with the nonblocking router
+     */
+    PUT_NB(true, false),
+    /**
+     * GetBlobInfo with the nonblocking router and check the blob info against what was put in.
+     */
+    GET_INFO_NB(true, false),
+    /**
+     * GetBlob with the nonblocking router and check the blob contents against what was put in.
+     */
+    GET_NB(true, false),
+    /**
+     * DeleteBlob with the nonblocking router
+     */
+    DELETE_NB(true, false),
+    /**
+     * GetBlobInfo with the nonblocking router. Expect an exception to occur because the blob should have already been
+     * deleted
+     */
+    GET_INFO_DELETED_NB(true, true),
+    /**
+     * GetBlob with the nonblocking router. Expect an exception to occur because the blob should have already been
+     * deleted
+     */
+    GET_DELETED_NB(true, true),
+    /**
+     * PutBlob with the coordinator-backed router
+     */
+    PUT_COORD(false, false),
+    /**
+     * GetBlobInfo with the coordinator-backed router and check the blob info against what was put in.
+     */
+    GET_INFO_COORD(false, false),
+    /**
+     * GetBlob with the coordinator-backed router and check the blob contents against what was put in.
+     */
+    GET_COORD(false, false),
+    /**
+     * DeleteBlob with the coordinator-backed router.
+     */
+    DELETE_COORD(false, false),
+    /**
+     * GetBlobInfo with the coordinator-backed router. Expect an exception to occur because the blob should have
+     * already been deleted.
+     */
+    GET_INFO_DELETED_COORD(false, true),
+    /**
+     * GetBlob with the coordinator-backed router. Expect an exception to occur because the blob should have already
+     * been deleted.
+     */
+    GET_DELETED_COORD(false, true),
+    /**
+     * Wait for the operation chain's blob ID to be reported as created on all replicas. Continue with the remaining
+     * actions in the operation chain afterwards.
+     */
+    AWAIT_CREATION(false, false),
+    /**
+     * Wait for the operation chain's blob ID to be reported as deleted on all replicas. Continue with the remaining
+     * actions in the operation chain afterwards.
+     */
+    AWAIT_DELETION(false, false);
+
+    final boolean nonBlockingRouter;
+    final boolean afterDelete;
+
+    OperationType(boolean nonBlockingRouter, boolean afterDelete) {
+      this.nonBlockingRouter = nonBlockingRouter;
+      this.afterDelete = afterDelete;
+    }
+  }
+
+  /**
+   * Describes an operation chain and provides useful metadata and operation results for testing.
+   */
+  static class OperationInfo {
```

Comment:
As discussed earlier today, this class is really not about info of an operation, but info of an operation chain. Please consider rename it to avoid confusion.